packages/nodes/src/ethereum.node.service.ts

```diff
@@ -20,116 +19,145 @@ import {
-         *  chainId: this.widgetConfig.chainId,
-         *  rpcUrl: WIDGET_CONFIG.STAND_CONFIG.rpcUrl,
-         *  defaultBalance: 100,
-         *  accounts: [{
-         *    account: '<secret_key>',
-         *    balance: 50
-         *  }]
-         *
-         * )
-         * ```
-         */
-        accounts: this.options.accounts,
-        defaultBalance: this.options.defaultBalance || 1000,
-      },
+  private verifyAnvil(): Promise<boolean> {
+    return new Promise((resolve) => {
+      const socket = new net.Socket();
+      socket.setTimeout(1000);
+      socket.once('error', () => resolve(false));
+      socket.once('timeout', () => resolve(false));
+      socket.connect(this.port, '127.0.0.1', () => {
+        socket.end();
+        resolve(true);
+      });
+    });
+  }
+
+  private startAnvil(rpcUrl: string) {
+    const args = [
+      `--fork-url=${rpcUrl}`,
+      '--block-time=2',
+      '--mnemonic=park pond parade curious ten impulse outdoor feel cousin party duck inherit',
+    ];
+    if (this.options.chainId) args.push(`--chain-id=${this.options.chainId}`);
+
+    const process = spawn('anvil', args, { stdio: 'pipe' });
+
+    process.stdout.once('data', (data: Buffer) => {
+      const text = data.toString();
+      this.logger.debug(`[Anvil STDOUT] ${text}`);
+
+      const keyMatches = [...text.matchAll(/\(\d+\)\s+(0x[a-fA-F0-9]{64})/g)];
+      keyMatches.forEach((match) => this.privateKeys.push(match[1]));
     });
 
-    await node.listen(this.options.port || 7545);
-    const nodeUrl = `http://127.0.0.1:${this.options.port || 7545}`;
-    const initialAccounts = await node.provider.getInitialAccounts();
-    const accounts: Account[] = Object.keys(initialAccounts).map((key) => {
-      return { address: key, secretKey: initialAccounts[key].secretKey };
+    process.stderr.on('data', (data) => {
+      this.logger.error(`[Anvil STDERR] ${data}`);
     });
-    await Promise.all(
-      accounts.map(async (account: Account) => {
-        await node.provider.send('evm_setAccountNonce', [
-          account.address,
-          '0x' + Math.floor(Math.random() * 9) + 1,
-        ]);
-      }),
-    );
 
-    node.on('close', async () => {
-      this.state = undefined;
+    process.on('close', (code) => {
+      this.logger.warn(`[Anvil Closed] Code ${code}`);
     });
-    this.state = { node, nodeUrl, accounts };
+
+    return process;
   }
 
-  getAccount(index = 0) {
-    return this.state.accounts[index];
+  async startNode() {
+    if (this.state) return;
+
+    const rpcUrl = this.options.rpcUrl;
+    if (!rpcUrl) throw new Error('RPC URL is required');
+
+    this.logger.debug('Starting Anvil node...');
+    if (!(await this.ensurePortAvailable(this.port))) {
+      this.logger.warn('Port already in use. Cleaning up before restart...');
+      await this.stopNode();
+    }
+
+    const process = this.startAnvil(rpcUrl);
+    const nodeUrl = `http://127.0.0.1:${this.port}`;
```

Comment:
lets keep host and port in the const? you are using few times host

Reply:
Fixed. Defined `this.port`as class var and pass it to anvil args